src/api/providers/kilocode.ts

```diff
@@ -174,17 +192,30 @@ export class KiloCodeHandler extends BaseProvider implements SingleCompletionHan
 		}
 	}
 
-	async completePrompt(prompt: string) {
-		let { id: modelId, temperature } = this.getModel()
+	async completePrompt(prompt: string, taskId?: string, checkpointNumber?: number) {
+		const { id: modelId, temperature } = this.getModel()
 
-		const message = await this.client.messages.create({
-			model: modelId,
-			max_tokens: ANTHROPIC_DEFAULT_MAX_TOKENS,
-			thinking: undefined,
-			temperature,
-			messages: [{ role: "user", content: prompt }],
-			stream: false,
-		})
+		// Prepare request options with headers
+		const requestOptions: { headers: Record<string, string> } = {
+			headers: {},
+		}
+
+		// Add idempotency key if task_id and checkpoint number are provided
+		if (taskId && checkpointNumber !== undefined) {
+			requestOptions.headers["idempotency-key"] = this.getIdempotencyKey(taskId, checkpointNumber)
+		}
```

Comment:
Maybe write this a bit more compactly? It feels like "a lot" in the way it's spread over so many lines, while in reality almost nothing is happening here. Code "weight" should feel proportional to the weight of the content.